acme/acme/client.py

```diff
@@ -768,12 +811,7 @@ def __init__(self, net, key, server):
             self.client = ClientV2(directory, net=net)
 
     def __getattr__(self, name):
-        if name in vars(self.client):
-            return getattr(self.client, name)
-        elif name in dir(ClientBase):
-            return getattr(self.client, name)
-        else:
-            raise AttributeError()
+        return getattr(self.client, name)
```

Comment:
Is this change because something was erroring? Note that it checks `vars` and `dir` on different classes -- it's meant to only call directly into shared methods, for our own checking.

Reply:
Well, the wrapper could not find the poll method when the delegate is ClientV2. And when I saw this method, I could not understand the advantage toward a proper delegate.

If I understand correctly, it makes necessary to implement the method on ClientBase, then override it on every child of ClientBase. And the purpose I suppose is to ensure that ClientBase always represents the common skeleton of the childs.

But it is not really Pythonic in my opinion, and not easy to understand. The result is a weird AttributeError that gives no clue of what is going on, while you would expect from a wrapper to just delegate to the underlying delagate object. 

A more Pythonic approach would be to implement on ClientBase the method as raising a NotImplementedError, that would lead straight forward the developper to the real error (we need to implement the method on all childs). Or using abc decorators.

I do not want to add a dummy poll method on ClientBase. The other solution would be to implement poll on ClientBase to use get, then override it on ClientV2 to use POST-as-GET. But it would be an infringement of what you said previously, as it would add a logic on ClientBase that is not relevant for ClientV2, without implementing the specific for Client.

So I think it is preferable to make a complete delegate like it did. Less code, on the right place, with Pythonic duck type approach.

Reply:
The point of this was to make an explicit forcing function to make sure that all the differences between V1 and V2 were captured, so that our testing didn't automatically pass through and just test one of the versions of some method. Making it a complete delegate is saying that we think this code is stable enough that we're unlikely to make that sort of mistake in the future. That may be the case.